### PR TITLE
Adding the capacity for an actor to spawn another actor.

### PR DESCRIPTION
**Pull request by fulmicoton**
Removed the supervising code in the actor handle.
Renames spawn_async_actor.

### CODE REVIEW

File: quickwit-indexing/src/actors/indexer.rs [hunk 1]
```diff
@@ -66,12 +70,32 @@ pub struct IndexerCounters {
     pub num_docs_in_split: u64,
 }
 
+impl IndexerCounters {
+    /// Returns the overall number of docs that went through the indexer (valid or not).
```
Comment (mosuka) [inline quickwit-indexing/src/actors/indexer.rs:74]: The document is the same as `num_invalid_docs()`.

File: quickwit-indexing/src/actors/indexer.rs [hunk 2]
```diff
@@ -2,7 +2,7 @@
 //  Copyright (C) 2021 Quickwit Inc.
 //
 //  Quickwit is offered under the AGPL v3.0 and as commercial software.
-//  For commercial licensing, contact us at hello@quickwit.io.
+//  For commercial li num_splits_emitted: () censing, contact us at hello@quickwit.io.
```
Comment (mosuka) [inline quickwit-indexing/src/actors/indexer.rs:5]: The license text is corrupted.

**Comment by mosuka** [comment]
@fulmicoton 
I left a comment, though it was a minor one.

File: quickwit-indexing/src/actors/indexer.rs (hunk 2 above)
Comment (fulmicoton) [reply quickwit-indexing/src/actors/indexer.rs]: good catch

**Comment by fulmicoton** [comment]
@mosuka, oops. I added a commit that was not meant for this PR. I removed it. (And apply your comment to the other commit)

**Comment by mosuka** [comment]
@fulmicoton oh. ok, thanks!

**Review by mosuka — APPROVED** [review]
looks good to me.

**Pull request merged**